### PR TITLE
Doc: Add @throws #369

### PR DESCRIPTION
**Pull request by LeonFLK**
## fixes https://github.com/KILTprotocol/ticket/issues/369
Improves the jsdoc coverage and adds @throw tags to all functions that throw errors, this branch is based on lw-154-constructor-checks, as this adds a lot of Error throwing functions.

Please provide feedback for the added documentation and request away!

Should the throw tags only represent the Errors the function explicitly throws or should we add the called functions thrown Errors too, especially in the case of our own verify functions, called mostly to check objects? @tjwelde 

## Checklist:

- [x] I have verified that the code works
- [x] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [x] I have [left the code in a better state](https://deviq.com/boy-scout-rule/)
- [x] I have documented the changes (where applicable)

### CODE REVIEW

**Comment by LeonFLK** [comment]
This will be much easier to review when https://github.com/KILTprotocol/sdk-js/pull/228 is merged

File: src/claim/Claim.utils.ts [hunk 1]
```diff
@@ -33,7 +33,7 @@ export function compress(claim: IClaim): CompressedClaim {
  *  Decompresses the [[Claim]] from storage and/or message.
  *
  * @param claim A compressesd [[Claim]] array that is reverted back into an object.
- *
+ * @throws When [[claim]] is not an Array or it's length is unequal 3.
```
Comment (tjwelde) [inline src/claim/Claim.utils.ts:36]: does this work, when using lowercase for `[[claim]]`?

File: src/ctype/CType.utils.ts [hunk 2]
```diff
@@ -54,6 +63,7 @@ export function getHashForSchema(schema: ICType['schema']): string {
  *  Compresses a [[CType]] schema for storage and/or messaging.
  *
  * @param cTypeSchema A [[CType]] schema object that will be sorted and stripped for messaging or storage.
+ * @throws When any of the four properties of the cTypeSchema are missing.
```
Comment (tjwelde) [inline src/ctype/CType.utils.ts:66]:
```suggestion
 * @throws When any of the four required properties of the cTypeSchema are missing.
```

File: src/identity/Identity.ts [hunk 3]
```diff
@@ -71,6 +74,7 @@ export default class Identity extends PublicIdentity {
     if (phrase) {
       if (phrase.trim().split(/\s+/g).length < 12) {
         // https://www.npmjs.com/package/bip39
+        // Actually only checks whether the phrase is too short.
         throw Error(`Phrase '${phrase}' too long or malformed`)
```
Comment (tjwelde) [inline src/identity/Identity.ts:78]: Please just change the error message

File: src/messaging/Message.ts [hunk 4]
```diff
@@ -130,6 +144,16 @@ export default class Message implements IMessage {
     }
   }
 
+  /**
+   * [STATIC] Decodes an [[IEncryptedMessage]] and returns the from it derived [[IMessage]].
+   *
+   * @param encrypted [[IEncryptedMessage]] to decode and build the [[IMessage]] from.
+   * @param receiver Address of the sender to verify the encrypted's signature against.
+   * @throws When decryption of the encrypted returns false.
```
Comment (tjwelde) [inline src/messaging/Message.ts:152]: "When encrypted message couldn't be decrypted"

File: src/quote/Quote.utils.ts [hunk 5]
```diff
@@ -39,7 +40,9 @@ export function compressCost(cost: ICostBreakdown): CompressedCostBreakdown {
 /**
  *  Decompresses the cost from storage and/or message.
  *
- * @param cost A compressesd cost array that is reverted back into an object.
+ * @param cost A compressed cost array that is reverted back into an object.
+ * @param cost A compressed cost array that is reverted back into an object.
```
Comment (tjwelde) [inline src/quote/Quote.utils.ts:44]: duplicate

File: src/ctype/CType.utils.ts (hunk 2 above)
Comment (LeonFLK) [reply src/ctype/CType.utils.ts]: Added

File: src/claim/Claim.utils.ts (hunk 1 above)
Comment (LeonFLK) [reply src/claim/Claim.utils.ts]: Capitalized

File: src/quote/Quote.utils.ts (hunk 5 above)
Comment (LeonFLK) [reply src/quote/Quote.utils.ts]: there were other duplicates, removed all

File: src/identity/Identity.ts (hunk 3 above)
Comment (LeonFLK) [reply src/identity/Identity.ts]: changed to "too short or malformed"

**Pull request merged**